pkg/statemetrics/mr_state_metrics.go

```diff
@@ -81,18 +82,20 @@ type MRStateRecorder struct {
 	log         logging.Logger
 	interval    time.Duration
 	managedList resource.ManagedList
+	gvk         schema.GroupVersionKind
 
 	metrics *MRStateMetrics
 }
 
 // NewMRStateRecorder returns a new MRStateRecorder which records the state of managed resources.
-func NewMRStateRecorder(client client.Client, log logging.Logger, metrics *MRStateMetrics, managedList resource.ManagedList, interval time.Duration) *MRStateRecorder {
+func NewMRStateRecorder(client client.Client, log logging.Logger, metrics *MRStateMetrics, managedList resource.ManagedList, gvk schema.GroupVersionKind, interval time.Duration) *MRStateRecorder {
```

Comment:
```suggestion
func NewMRStateRecorder(c client.Client, log logging.Logger, m *MRStateMetrics, ml resource.ManagedList, gvk schema.GroupVersionKind, interval time.Duration) *MRStateRecorder {
```

Nit: With six arguments this function signature is getting pretty long. Some of the variable names could be shorter without meaningful loss of information (e.g. `c` is fine when you can see its a client from the type `client.Client`).

https://github.com/crossplane/crossplane/tree/master/contributing#dont-wrap-function-signatures